### ISSUE TITLE
Remove `peripherals::` prefix

### ISSUE THREAD

**Issue by JuliDi** [adamgreig/assign-resources#8]
I was wondering whether there is a specific reason for doing `peripherals::$resource_field`  instead of just `$resource_field`.
The user will have to `use embassy_stm32::peripherals` anyway and could just as well do `use embassy_stm32::peripherals::*` (or not?).

Removing this prefix would make the macro a lot more flexible and allow to include arbitrary resources into it, e.g., an instance of a driver, some array, etc.

**Comment by adamgreig** [comment]
The built in reference to `peripherals` is one of the only things that makes this particularly embassy-specific too, I don't love it but I do love how much neater it makes writing out the macro call. I'm not a huge fan of the wildcard import but I think it would probably work OK.

Perhaps removing it and letting the user either import `peripherals::*` or prefix each one with `peripherals::` in the call is the best option. It makes the macro a bit less "magic" too.

**Comment by JuliDi** [comment]
I agree, it's really neat how you can just type the name of the peripheral. So that's a tradeoff in the end.
However, for new users it might be an unexpected surprise that nothing other than embassy peripherals (or at least nothing that's not in a peripherals namespace) work inside the macro.

In the end you can always copy the macro into your own library and customize it the way you need, so this is not a real issue.
